content/2017/03/30/safe-get.rst

```diff
@@ -131,161 +135,104 @@ In demo code, people often write it like this:
-for handling functions defined elsewhere
-(e.g. third-party tools)
-that won't have guaranteed access
-to the functions being called.
-Using ``get-function``,
-we can pass a first-class function
-to a third-party toolkit,
-without worrying about differences in namespace.
-So, a more accurate demonstration might be:
+The ``call`` option is mostly useful
+for handling functions defined elsewhere.
+Since we don't have guaranteed access
+to the functions being called in our third-party app,
+``get-function`` allows us to capture and pass-in
+functions from a different scope.
+
+A more accurate demonstration might be:
 
 .. code:: scss
 
   // third-party.scss
-  @mixin three($function) {
-    .three {
-      width: call($function, 3);
-    }
+  @mixin three-wide($function) {
+    width: call($function, 3);
   }
 
   // my-local.scss
-  @import 'susy';
-  @include three(get-function('susy.span'));
-
-This still creates a problem for toolkits and frameworks
-(like `Susy`_)
-that already use ``call()`` internally
-to handle user input.
-How do we support old and new versions of Sass,
-while allowing users to pass in
-either strings or first-class functions?
+  @import 'susy' as 'susy';
 
-`Kaelig provides one solution`_
-in a great article with more details.
-It's a good start,
-but it doesn't cover all the use cases I need.
-What if users pass in a first-class function
-that they've already captured –
-as they likely should in Sass ``3.5+``?
-Here's my slightly-expanded solution.
-
-.. _Susy: http://susy.oddbird.net
-.. _Kaelig provides one solution: https://medium.com/@kaelig/sass-first-class-functions-6e718e2b5eb0
+  $span-function: get-function('susy.span');
+  @include three-wide($span-function);
 
+So how do we support old and new versions of Sass,
+while allowing users to pass in
+either strings or first-class functions?
 
-Safe ``get-function``
----------------------
-
-We need to use ``get-function()`` in new versions of Sass,
-but we can't use it in old versions.
-We also don't want to use ``get-function()``
-on a function we've already got.
-That gives us several options we have to cover
-in our ``safe-get-function()``.
-
-- If the user passes in a string,
-  and we're using an older version of Sass
-  => *do nothing*.
-- If the user passes in a string,
-  and we're using a newer versions of Sass
-  => use *get-function*.
-- If the user passes in a first-class function,
-  we can assume we're using the latest Sass version
-  => *do nothing*.
-
-The result looks something like this:
+The short answer is: we don't*.
+Our users will make that change on their end,
+when they upgrade.
+
+
+Don't Worry About It
+--------------------
+
+`Kaelig provides one solution`_,
+and I initially meant to improve on it
+with more robust features.
+**You don't need any of that.**
+
+After a long conversation with Chris Eppstein,
+one of the Sass language designers,
+it's clear that the entire upgrade path
+should be handled by users,
+not toolkit developers.
```

Comment:
I admit to not knowing much about the subject matter here, but this sentence seems to beg the question. Why is it clear?